### PR TITLE
Handle errors while inviting other memebers

### LINKED ISSUE #552: Frontend: Add validation before sending invite to other team members
- [ ] Handle errors (email required & email valid ) with ng-messages as it is done in auth pages.
- [x] Show notification on invitation success after closing the modal of email invite (depend on issue #482 )

### PR DESCRIPTION
**Pull request by gautamjajoo**
Fixes #552 
Similar to #1542 . Validation done using `ng-pattern`.

screenshot-from-2018-11-20-19-34-34
screenshot-from-2018-11-20-19-34-45

@deshraj @RishabhJain2018

### CODE REVIEW

**Comment by gautamjajoo** [comment]
@deshraj Please review this too.

**Comment by deshraj** [comment]
@RishabhJain2018 will take care of this. :)

**Comment by gautamjajoo** [comment]
@RishabhJain2018 Please review this too!.. :)

**Comment by deshraj** [comment]
@RishabhJain2018 can you please review this PR whenever you get time?

**Comment by RishabhJain2018** [comment]
@gautamjajoo Can you please resolve the conflicts?

**Comment by gautamjajoo** [comment]
@RishabhJain2018 Updated! :)

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 1]
```diff
@@ -7,9 +7,14 @@
         .module('evalai')
         .controller('ChallengeHostTeamsCtrl', ChallengeHostTeamsCtrl);
 
+    ChallengeHostTeamsCtrl.$inject = ['utilities', 'loaderService', '$state', '$http', '$scope', '$rootScope', '$mdDialog'];
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: @gautamjajoo Why are you adding this again?

**Comment by gautamjajoo** [comment]
Opening a new PR in some time.

**Pull request closed without merging**